### PR TITLE
feat(tools/respec2html): add --localhost option for http server

### PR DESCRIPTION
**Pull request by marcoscaceres**
Quick straw-person for #3292 for discussion.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
   },
   "dependencies": {
     "colors": "^1.4.0",
+    "express": "^4.17.1",
```
Comment (sidvishnoi) [inline package.json:85]: First thought before anything else: overkill for a static server! 😆

File: tools/respec2html.js [hunk 2]
```diff
@@ -43,7 +48,9 @@ cli
   .option("-w, --haltonwarn", "Abort if ReSpec generates warnings.", false)
   .option("--disable-sandbox", "Disable Chromium sandboxing if needed.", false)
   .option("--devtools", "Enable debugging and show Chrome's DevTools.", false)
-  .option("--verbose", "Log processing status to stdout.", false);
+  .option("--verbose", "Log processing status to stdout.", false)
+  .option("--localhost", "Spin up a local server to peform processing.", false)
```
Comment (sidvishnoi) [inline tools/respec2html.js:52]: Maybe we can support something like `--localhost=optionalPort` instead of two options.

File: package.json (hunk 1 above)
Comment (marcoscaceres) [reply package.json:85]: Yeah, was thinking the same... at the same time, it's nice to use. Do you have a suggestion for a good alternative. Generally, I use https://www.npmjs.com/package/http-server but it doesn't look like it has an API.

File: tools/respec2html.js (hunk 2 above)
Comment (marcoscaceres) [reply tools/respec2html.js:57]: Could do... need to check how to set up that kind of option.

File: package.json (hunk 1 above)
Comment (sidvishnoi) [reply package.json:85]: I used `serve-handler` in spec-prod

**Comment by sidvishnoi** [comment]
Maybe we can use local server as default behavior?

**Comment by marcoscaceres** [comment]
> Maybe we can use local server as default behavior?

I'm not comfortable with spinning up a web sever on a user's machine by default, tbh. Even if it's just for a few seconds. "The Web" is scary.

File: tools/respec2html.js (hunk 2 above)
Comment (marcoscaceres) [reply tools/respec2html.js:68]: Concluded to keeping option separate.

File: tools/respec2html.js [hunk 3]
```diff
@@ -43,7 +64,9 @@ cli
...
```
Comment (sidvishnoi) [inline tools/respec2html.js:68]: This message some a room for improvements. It looks a bit misleading right now.

File: tools/respec2html.js [hunk 4]
```diff
@@ -2,6 +2,22 @@
...
+    req,
+    res,
+    finalhandler(req, res, {
+      onerror(err) {
+        console.error(err.stack);
+      },
+    })
+  );
+});
+const path = require("path");
```
Comment (sidvishnoi) [inline tools/respec2html.js:20]: Can we wrap these in a `StaticServer` class?

Comment (sidvishnoi) [reply tools/respec2html.js]: Done with 2158b46

File: tools/respec2html.js (hunk 3 above)
Comment (marcoscaceres) [reply tools/respec2html.js:121]: Feel free to change.... the current wording makes sense to me so I couldn't come up with anything better 😢.

**Review by marcoscaceres — COMMENTED** [review]
The new class for the server looks good to me. You will need to self approve to merge it, but happy to review any new changes you make.

File: tools/respec2html.js (hunk 3 above)
Comment (sidvishnoi) [reply tools/respec2html.js:121]: Can't think of any better either. Right now, it sounds like the server will perform processing..

File: tools/respec2html.js [hunk 5]
```diff
@@ -38,6 +42,49 @@ class Logger {
   }
 }
 
+class StaticServer {
+  /**
+   * @param {number} port
+   * @param {string} source
+   */
+  constructor(port, source) {
```
Comment (marcoscaceres) [inline tools/respec2html.js:50]: I wonder if we should split construction from source validation... they seem to be different concerns.

**Review by marcoscaceres — COMMENTED** [review]
Just a minor thing for the constructor

File: tools/respec2html.js (hunk 3 above)
Comment (marcoscaceres) [reply tools/respec2html.js]: That's not entirely untrue... if you squint just right.

File: tools/respec2html.js (hunk 5 above)
Comment (sidvishnoi) [reply tools/respec2html.js:50]: It encapsulates related logic well enough though..

Comment (marcoscaceres) [reply tools/respec2html.js:50]: Up do you. I don't mind either way.

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**